libsolidity/codegen/CompilerContext.cpp

```diff
@@ -227,6 +227,13 @@ void CompilerContext::injectVersionStampIntoSub(size_t _subIndex)
 	sub.injectStart(fromBigEndian<u256>(binaryVersion()));
 }
 
+void CompilerContext::injectMetadataHashIntoSub(size_t _subIndex, h256 const& _metadataHash)
+{
+	eth::Assembly& sub = m_asm->sub(_subIndex);
+	sub.injectStart(Instruction::POP);
+	sub.injectStart(u256(_metadataHash));
```

Comment:
When the number `_metadataHash` is accidentially small, this could produce `PUSH31` instead of `PUSH32`.